maxnorm option type

I see 5 instances of this warning in the log of a build of version 1.6.12:
```
In file included from main.cpp:37:
../../src/zsolve/DefaultController.hpp: In member function ‘void _4ti2_zsolve_::DefaultController<T>::log_maxnorm(_4ti2_zsolve_::Algorithm<T>*, bool)’:
../../src/zsolve/DefaultController.hpp:255:74: warning: comparison of constant ‘2’ with boolean expression is always false [-Wbool-compare]
  255 |         if ((m_options.maxnorm () == 1 && final) || m_options.maxnorm () == 2)
      |                                                     ~~~~~~~~~~~~~~~~~~~~~^~~~
```

The warning means that GCC is optimizing away the check for equality with the number 2, because the type of `m_options.maxnorm()` is `bool`, which can only be false (0) or true (1).  If maxnorm can really be 2, then its type should be changed from bool to int.  If maxnorm cannot really be 2, then the comparisons with 2 should be removed.

--- a/src/zsolve/DefaultController.hpp
+++ b/src/zsolve/DefaultController.hpp
@@ -252,7 +252,7 @@ public:
 
     void log_maxnorm (Algorithm <T> * algorithm, bool final)
     {
-        if ((m_options.maxnorm () == 1 && final) || m_options.maxnorm () == 2)
+        if (m_options.maxnorm () && final)
         {
             VectorArray <T> maxnorm (algorithm->get_result_variables ());
             T norm = algorithm->extract_maxnorm_results (maxnorm);

